Mathlib/Topology/Separation.lean

```diff
@@ -909,6 +911,38 @@ instance (priority := 100) ConnectedSpace.neBot_nhdsWithin_compl_of_nontrivial_o
     contra (compl_union_self _) (Set.nonempty_compl_of_nontrivial _) (singleton_nonempty _)
   simp [compl_inter_self {x}] at contra
 
+theorem SeparationQuotient.t1Space_iff : T1Space (SeparationQuotient X) ↔ R0Space X := by
+  rw [r0Space_iff, Symmetric]
+  have : ∀ x y : X, x ⤳ y → mk x ⤳ mk y := by
+    intro x y h
+    apply (Inducing.specializes_iff ?hf).mpr h
+    exact SeparationQuotient.inducing_mk
+  have := (t1Space_TFAE (SeparationQuotient X)).out 0 9
+  rw [this]
+  constructor
+  · intro h
+    intro x y xspecy
+    have : mk x ⤳ mk y := (Inducing.specializes_iff ?hf).mpr xspecy
+    have : mk x = mk y := h this
+    apply (Inducing.specializes_iff ?hf).mp
+    rw [this]
+  · intro h sx sy sxspecsy
+    rcases Quotient.exists_rep sx with ⟨ x, hx ⟩
+    rcases Quotient.exists_rep sy with ⟨ y, hy ⟩
+    replace hx : mk x = sx := hx
+    replace hy : mk y = sy := hy
+    rw [← hx, ← hy] at sxspecsy
+    have xspecy : x ⤳ y := (Inducing.specializes_iff ?hf).mp sxspecsy
+    have yspecx : y ⤳ x := h xspecy
+    have : Inseparable x y := by
+      rw [inseparable_iff_specializes_and]
+      exact ⟨ xspecy, yspecx ⟩
+    have : mk x = mk y := by
+      rw [mk_eq_mk]
+      exact this
+    rw [← hx, ← hy]
+    exact this
```

Comment:
```suggestion
theorem SeparationQuotient.t1Space_iff : T1Space (SeparationQuotient X) ↔ R0Space X := by
  rw [r0Space_iff, Symmetric]
  have hind : Inducing (mk : X → SeparationQuotient X) := SeparationQuotient.inducing_mk
  have := (t1Space_TFAE (SeparationQuotient X)).out 0 9
  rw [this]
  constructor
  · intro h x y xspecy
    rw [← Inducing.specializes_iff hind] at *
    rw [h xspecy]
  · intro h sx sy sxspecsy
    rcases Quotient.exists_rep sx with ⟨ x, rfl ⟩
    rcases Quotient.exists_rep sy with ⟨ y, rfl ⟩
    suffices Inseparable x y by rwa [Quotient.sound]
    rw [inseparable_iff_specializes_and]
    have xspecy : x ⤳ y := (Inducing.specializes_iff hind).mp sxspecsy
    have yspecx : y ⤳ x := h xspecy
    exact ⟨ xspecy, yspecx ⟩
```
I am no means an expert, but I thought I'd leave some general advice.
When using `(Inducing.specializes_iff ?hf).mp sxspecsy`, the `?hf` becomes a new goal. I'm honestly not sure how `have : ∀ x y : X, x ⤳ y → mk x ⤳ mk y` was allowing lean to close that goal on its own, but in general this is poor style. I replaced it with `have hind : Inducing (mk : X → SeparationQuotient X) := SeparationQuotient.inducing_mk` which I used in place of `?hf`.
Instead of `have : mk x ⤳ mk y := (Inducing.specializes_iff ?hf).mpr xspecy` and adding the context, it's usually best to use `simp at xspecy` or `rw at xspecy` to massage it into the form you need. In this case, I used `rw at *` to rewrite both the goal and `xspecy`.
With `rcases Quotient.exists_rep`, I used `rfl` to rewrite `sx` and `sy` and avoided rewriting to `mk`.
From there, I just reorganized the argument to follow more linearly.

Reply:
Thanks! I'll walk through this myself and use it (I also have some feedback in Zulip that I'll work through as well).